code/modules/mining/machine_processing.dm

```diff
@@ -156,6 +169,89 @@
 	src.updateUsrDialog()
 	return
 
+/obj/machinery/mineral/processing_unit_console/proc/print_report(var/mob/living/user)
+	if(!inserted_id)
+		to_chat(user, span("warning", "No ID inserted. Cannot digitally sign."))
+		return
+	if(!input_mats.len && !output_mats.len && !alloy_mats)
+		to_chat(user, span("warning", "There is no data to print."))
+		return
+
+	var/obj/item/weapon/paper/P = new /obj/item/weapon/paper(user.loc)
+	var/date_string = worlddate2text()
+	idx++
+
+	var/form_title = "Form 0600 - Mining Yield Declaration"
+	var/dat = "<small><center><b>NanoTrasen Inc.<br>"
+	dat += "Civilian Branch of Operation</b><br><br>"
+
+	dat += "Form 0600<br> Mining Yield Declaration</center><hr>"
+	dat += "Facility: NSS Aurora<br>"
+	dat += "Date: [date_string]<br>"
+	dat += "Index: [idx]<br><br>"
+
+	dat += "Miner(s): <span class=\"paper_field\"></span><br>"
+	dat += "Ore Yields: <br>"
+
+	dat += "<table>"
+	for(var/material/OM in output_mats) {
```

Comment:
Why are squigglies used here and below? It's not normal style outside of macros.